src/utils/object/omit.ts

```diff
@@ -1,8 +1,10 @@
+import { deleteNestedKey } from "./deleteNestedKey";
+
 export const omit = <T extends object, K extends [...(keyof T)[]]>(
   obj: T,
   params: K,
 ) => {
   const newObj = { ...obj };
-  params.forEach((param) => delete newObj[param]);
+  deleteNestedKey(newObj, params as string[]);
   return newObj as Omit<T, K[number]>;
```

Comment:
The typing here is no longer valid it does not handle nested keys.

Reply:
I think it might makes sense to keep these utility functions separate since one handles simple key value maps and the other handles deeply nested objects. What do you think?